### PR TITLE
Correct treatment of the unseen labels in the training set

### PR DESCRIPTION
**Pull request by aloukina**

### CODE REVIEW

File: tests/test_classification.py [hunk 1]
```diff
@@ -323,3 +323,72 @@ def test_adaboost_predict():
...
+
+
+def check_results_with_unseen_labels(res, n_labels, new_label_list):
+    (confusion_matrix,
+     score,
+     result_dict,
+     model_params,
+     grid_score) = res
+
+    # check that the new metric is included into the results
```
Comment (desilinguist) [inline tests/test_classification.py]: new "metric"?

File: tests/test_classification.py [hunk 2]
```diff
@@ -323,3 +323,72 @@ def test_adaboost_predict():
...
+    learner = Learner('SVC')
+    learner.train(train_fs, grid_search=False)
+    res = learner.evaluate(test_fs)
+    yield check_results_with_unseen_labels, res, 4, [15]
+    yield assert_almost_equal, res[1], 0.3
+
+
+def test_all_new_labels_in_test():
+    """
+    Test classification with all labes in test set unseen
```
Comment (desilinguist) [inline tests/test_classification.py]: "labes" -> "labels"

**Comment by desilinguist** [comment]
Looks pretty good. Just a trivial typo.

File: tests/test_classification.py (hunk 1 above)
Comment (aoifecahill) [inline tests/test_classification.py]: do you mean label instead of metric?

**Comment by desilinguist** [comment]
@aloukina you need to merge in `master` as well.

**Comment by desilinguist** [comment]
It this ready, @aloukina?

**Comment by aloukina** [comment]
See my comment under #271 - we need to make sure that the labels in the confusion matrix are assigned correctly when the new labels fall between the old labels. Currently my branch generates confusion matrix with new labels at the end of the matrix. I have not looked at that part of the code yet to see whether we need to fix this here or there.

**Comment by desilinguist** [comment]
I think that should be fixed in #271 after merging this one. Your PR by itself is ready right?

**Comment by aloukina** [comment]
yes

**Comment by desilinguist** [comment]
@aoifecahill since you filed this issue, is there some kind of data you can test on to make sure this is working as expected?

**Comment by aoifecahill** [comment]
Yep, will do.

**Comment by aoifecahill** [comment]
Works on my dataset :+1:

**Pull request merged**